City-of-Helsinki/haitaton-backend
# HAI-1638 Remove GeometriaController

# Description

The concept of hanke-geometria has been superceded by the concept of hankealue. One of the old geometria-endpoints was still up, but unused.

Remove the remaining geometria-endpoint, and the since it's the last method in the controller, the whole controller. Also remove the loadGeometriat method, which wasn't used anywhere else.

Refactor HankeControllerSecurityTests to test single calls separately to make it possible to see if any unnecessary stubbing of mocks is done.

Refactor HankeControllerITests
- Divide the tests for different endpoints to different inner classes.
- Move HankeControllerHankeEditingDisabledITests to a separate file.
- Move the unauthorized tests from HankeControllerSecurityTests to
  HankeControllerITests.
- Remove the positive tests from HankeControllerSecurityTests, since
  they were duplicates for tests in HankeControllerITests.
  - Delete HankeControllerSecurityTests, since it was now empty.
- Use verifySequence instead of many separate verify -blocks.
- Introduce a ResultMatcher for checking that a response is a specific
  HankeError, based on a private method in HankeControllerSecurityTests.
  - Use it in all tests in HankeControllerSecurityTests
  - This should be used more in tests. Just checking for a status might
    lead to false positives, since we don't know what caused the status.
- Disable a test for checking that a perustaja without email is not
  allowed. The tested functionality doesn't seem to be implemented. It
  passed earlier, because the test was broken.

### Jira Issue: https://helsinkisolutionoffice.atlassian.net/browse/HAI-1638

## Type of change
